### PR TITLE
GH-45079: [FlightRPC][C++] Deprecate InitializeFlightUcx before removing UCX

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

We want to remove the UCX experiment as discussed on the issue:
- https://github.com/apache/arrow/issues/43296

Based on the discussions of the PR removal we have decided to deprecate it before.

### What changes are included in this PR?

Deprecation of InitializeFlightUcx

### Are these changes tested?

No, I don't think we test `ARROW_WITH_UCX` on CI.

### Are there any user-facing changes?

Yes, the PR deprecated Flight UCX
* GitHub Issue: #45079

### CODE REVIEW

**Review by lidavidm — APPROVED** [review]

File: docs/source/format/Flight.rst [hunk 1]
```diff
@@ -333,9 +333,13 @@ schemes for the given transports:
 +----------------------------+--------------------------------+
 | (reuse connection)         | arrow-flight-reuse-connection: |
 +----------------------------+--------------------------------+
-| UCX_ (plaintext)           | ucx:                           |
+| UCX_ (plaintext) (1)       | ucx:                           |
 +----------------------------+--------------------------------+
 
+Notes:
+
+* \(1) Flight UCX transport has been deprecated on the 19.0.0 release.
```
Comment (kou) [inline docs/source/format/Flight.rst:341]: Can we add a link to https://arrow.apache.org/docs/dev/format/DissociatedIPC.html ?

**Review by kou — APPROVED** [review]
+1

**Comment by raulcd** [comment]
@github-actions crossbow submit preview-docs

File: docs/source/format/Flight.rst (hunk 1 above)
Comment (raulcd) [reply docs/source/format/Flight.rst:341]: Done, see: http://crossbow.voltrondata.com/pr_docs/45080/format/Flight.html#location-uris

**Pull request merged**